src/Tools/dotnet-monitor/CollectionRules/Configuration/CustomShortcutsConfigureNamedOptions.cs

```diff
@@ -0,0 +1,225 @@
+                bool triggerInsertion = InsertCustomTriggerIntoTrigger(options, name);
+                bool filterInsertion = InsertCustomFiltersIntoFilterList(options, name);
+                bool limitInsertion = InsertCustomLimitIntoLimit(options, name);
+
+                // If any insertions fail, clear out the options so that the rule will fail Validation
+                if (!(actionInsertion && triggerInsertion && filterInsertion && limitInsertion))
+                {
+                    options.Trigger = null;
+                    options.Actions.Clear();
+                    options.Filters.Clear();
+                    options.Limits = null;
+                }
+            }
+        }
+
+        private bool InsertCustomActionsIntoActionList(CollectionRuleOptions ruleOptions, string name)
+        {
+            bool boundAllActions = false;
+            int index = 0;
+
+            while (!boundAllActions)
+            {
+                IConfigurationSection section = _configuration.GetSection(ConfigurationPath.Combine(nameof(RootOptions.CollectionRules), name, nameof(CollectionRuleOptions.Actions), index.ToString(CultureInfo.InvariantCulture)));
+                if (section.Exists() && !string.IsNullOrEmpty(section.Value))
+                {
+                    if (!InsertCustomShortcut(_shortcutOptions.CurrentValue.Actions, section.Value, ruleOptions.Actions, index))
+                    {
+                        return false;
+                    }
+                }
+                else if (!section.Exists())
+                {
+                    boundAllActions = true;
+                }
+
+                ++index;
+            }
+
+            return true;
+        }
+
+        private bool InsertCustomTriggerIntoTrigger(CollectionRuleOptions ruleOptions, string name)
+        {
+            IConfigurationSection section = _configuration.GetSection(ConfigurationPath.Combine(nameof(RootOptions.CollectionRules), name, nameof(CollectionRuleOptions.Trigger)));
+            if (section.Exists() && !string.IsNullOrEmpty(section.Value))
+            {
+                if (!ValidateCustomShortcutExists(_shortcutOptions.CurrentValue.Triggers, section.Value))
+                {
+                    return false;
+                }
+
+                CollectionRuleTriggerOptions options = _shortcutOptions.CurrentValue.Triggers[section.Value];
+                ruleOptions.Trigger = options;
+            }
+
+            return true;
+        }
+
+        private bool InsertCustomFiltersIntoFilterList(CollectionRuleOptions ruleOptions, string name)
+        {
+            bool boundAllFilters = false;
+            int index = 0;
+
+            while (!boundAllFilters)
+            {                
+                IConfigurationSection section = _configuration.GetSection(ConfigurationPath.Combine(nameof(RootOptions.CollectionRules), name, nameof(CollectionRuleOptions.Filters), index.ToString(CultureInfo.InvariantCulture)));
```

Comment:
Why not get all the children for the element and iterate through them instead of maintaining and testing an index?